rca/research/models.py

```diff
@@ -1,6 +1,313 @@
+            ],
+            heading=_("Introduction"),
+        ),
+        MultiFieldPanel(
+            [InlinePanel("research_spaces", label="Research spaces")],
+            heading="Research spaces",
+        ),
+        MultiFieldPanel(
+            [InlinePanel("research_opportunities", label="Research opportunities")],
+            heading="Research opportunities",
+        ),
+        MultiFieldPanel(
+            [
+                FieldPanel("more_research_centre_content_title"),
+                InlinePanel("research_news", label="Research news"),
+            ],
+            heading="More research centre content",
+        ),
+        MultiFieldPanel(
+            [
+                FieldPanel("staff_title"),
+                InlinePanel("related_staff", label="staff"),
+                FieldPanel("staff_link"),
+                FieldPanel("staff_link_text"),
+            ],
+            heading="Research Centre Staff",
+        ),
+        StreamFieldPanel("related_links"),
+    ]
+    key_details_panels = [
+        MultiFieldPanel(
+            [
+                FieldPanel("primary_staff_name"),
+                FieldPanel("primary_staff_url"),
+                FieldPanel("primary_staff_role"),
+            ],
+            heading="Primary staff member",
+        ),
+        MultiFieldPanel(
+            [
+                FieldPanel("centre_address"),
+                FieldPanel("centre_tel"),
+                FieldPanel("centre_email"),
+            ],
+            heading="Centre contact information",
+        ),
+        FieldPanel("twitter_username"),
+    ]
+
+    edit_handler = TabbedInterface(
+        [
+            ObjectList(content_panels, heading="Content"),
+            ObjectList(key_details_panels, heading="Key details"),
+            ObjectList(BasePage.promote_panels, heading="Promote"),
+            ObjectList(BasePage.settings_panels, heading="Settings"),
+        ]
+    )
+
+    def get_child_projects(self):
+        """
+        Returns a queryset of all child ProjectPages of this page
+        TODO: This needs finishing once we have the project pages built
+        Also a bit unsure on the template changes I've had to do
+        """
+        projects = (
+            ProjectPage.objects.live()
+            .public()
+            .descendant_of(self, inclusive=True)
+            .select_related("listing_image")
+        )
+        child_projects = []
+        for page in projects:
+            page = page.specific
+            child_projects.append(
+                {
+                    "title": page.title,
+                    "link": page.url,
+                    "image": page.listing_image,
+                    "description": page.listing_summary,
+                }
+            )
+        return child_projects
+
+    def get_research_spaces(self):
+        research_spaces = []
+        for value in self.research_spaces.select_related("page"):
+            if value.page.live:
```

Comment:
The `RelatedPage.page` field can be `null`, which would cause an `AttributeError` here if it were. https://github.com/torchbox/rca-wagtail-2019/blob/master/rca/utils/models.py#L98

I don't think `RelatedPage.page` should be nullable. Its `on_delete=models.CASCADE` so a null value is less likely, but might be worth changing to be safe.

Reply:
I understand this, If I remove the related page here, everything works fine. Are you suggesting that the I change the RelatedPage.page so it's not nullable? That sound like a safe move, my concern is handling the default values for existing entries, would it just be a blank string because blank=True?

Reply:
You'd also remove `blank=True` (so it'd use the default `blank=False`).

Yes ideally `RelatedPage.page` would be not nullable, but you're right in that default values for existing records could be an issue as it'll want a valid page id.

You could maybe have `if value.page and value.page.live:` instead

Reply:
Yea good shout - I'd still like to change that related page null though, seems to scare me having it hanging around. I'll add in a check for the page object too and create a follow issue for this https://github.com/torchbox/rca-wagtail-2019/issues/378